charts/nx-agents/templates/deployment.yaml

```diff
@@ -0,0 +1,82 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nx-cloud-workflow-controller
+  namespace: {{ .Values.global.namespace | quote }}
+  labels:
+    {{- include "nxCloud.app.labels" . | indent 4 }}
+  {{- if .Values.controller.deployment.annotations }}
+  annotations:
+    {{- toYaml .Values.controller.deployment.annotations | nindent 4 }}
+  {{- end
+spec:
+  selector:
+    matchLabels:
+      app: nx-cloud-workflow-controller
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: nx-cloud-workflow-controller
+    spec:
+      securityContext:
+        runAsNonRoot: true
+      containers:
+        - command:
+          - /nx-cloud-workflow-controller
+          - --namespace={{ .Values.global.namespace }}
+          - --in-cluster=true
+          {{- range $key, $value := .Values.executor.env }}
+          - --executor-env={{ $key }}={{ $value }}
+          {{- end }}
+          - --workflow-service-address=http://nx-cloud-workflow-controller-service:9000
+          image: {{ include "nxCloud.images.workflowController.image" . }}
+          imagePullPolicy: {{ .Values.controller.image.pullPolicy | quote }}
+          {{- if .Values.controller.resources }}
+          resources: {{- toYaml .Values.controller.resources | nindent 12 }}
+          {{- end }}
+          name: nx-cloud-workflow-controller
+          securityContext:
+            runAsUser: 1000
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - 'ALL'
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: {{ .Values.controller.deployment.port }}
+            initialDelaySeconds: 15
+            periodSeconds: 20
+          readinessProbe:
+            httpGet:
+              path: /readyz
+              port: {{ .Values.controller.deployment.port }}
+            initialDelaySeconds: 5
+            periodSeconds: 10
+          {{- if gt (len .Values.controller.env) 0 }}
+          env:
+          {{- range $key, $value := .Values.controller.env }}
+          - name: {{ $key | quote }}
+            value: {{ $value | quote }}
+          {{- end }}
+          {{- with .Values.secret }}
+          {{- if .awsS3AccessKeyId }}
+          {{- if .name }}
+          - name: AWS_S3_ACCESS_KEY_ID
+            valueFrom:
+              secretKeyRef:
+                name: {{ .name }}
+                key: {{ .awsS3AccessKeyId }}
+          - name: AWS_S3_SECRET_ACCESS_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ .name }}
+                key: {{ .awsS3SecretAccessKey }}
```

Comment:
we'll probably need some kind of role based access rather than id/key, but we can discuss that in the next few weeks

Reply:
I'm pretty sure that our controller/executor are similar to nx-api in that they need these credentials to speak with s3-compatible storage. 

We currently do use workload identity for GCP and can likely support the same. For anything else we'd need to look into workload identity pool providers for each platform. Wiring up the SA bindings is possible but will take a bit longer as we do not generally test this sort of thing aside from GCP